### DISCUSSION TITLE
Steamworks flat interface overhaul

### DISCUSSION THREAD

**Discussion by zpostfacto** [Thinkofname/steamworks-rs#30] [General]
I've just updated the Steamworks SDK.  Version 1.48 includes a pretty big overhaul to the flat interface, and a new windows-event-like manual callback dispatch mechanism.  The goal is to make binding layers like this easier to write.  Less manual maintenance, and less gross hackery under the hood.

When you get a sec, please take a look at it, and if you have any questions or complaints, email me.

**Comment by Scetch** [comment]
This is pretty exciting to hear :)

**Comment by Thinkofname** [comment]
Thanks for the work should make things a lot easier. Few things I've noticed so far:

- `SteamAPI_ISteamNetworking_ReadP2PPacket` Still has a `CSteamID` parameter
- `CSteamID/CGameID` could be helpful to have a way to access the methods of these types when using the raw uint64 versions.
- `SteamTVRegion_t`/`ESteamTVRegionBehavior` seem to be missing from the headers? They are used in the flat interface and are in the steam_api.json but not the headers.
- `SteamGameServer_Init` is an inline function so can't be called from bindings unlike `SteamAPI_Init`

**Comment by zpostfacto** [comment]
Thanks!  I'll look into those things and see what I can do.
